pages/common/uuencode.md

```diff
@@ -0,0 +1,16 @@
+# uuencode
+
+> Encode binary files into ASCII for transport via mediums that only support simple ASCII encoding.
+> More information: <https://manned.org/uuencode>.
+
+- Encode a file and print the result to stdout:
+
+`uuencode {{path/to/input_file}} {{output_file_name_after_decoding}}`
+
+- Encode a file and and write the result to a file:
+
+`uuencode -o {{path/to/output_file}} {{path/to/input_file}} {{output_file_name_after_decoding}}`
+
+- Encode a file in Base64 instead of the tradition uuencode encoding and and write the result to a file:
+
+`uuencode -m -o {{path/to/output_file}} {{path/to/input_file}} {{output_file_name_after_decoding}}`
```

Comment:
Is there a reason you don't use `--base64`? Also `-o` isn't listed in the `manned.org` page.

Reply:
Yes, there are two.

For one, I didn't know about it.

For two, one of the pains about Macs is that macOS _has_ to come bundled with the _most_ ancient and ritarded versions of every piece of software.  So my `uuencode` is BSD, and its manpage was last updated in 2002.  It only supports short options, so `--base64` is entirely out of the question.

I think we should move this page to `osx/`.

<details>
<summary>BSD uuencode's man page, in case you want to laugh with me</summary>

```man
UUENCODE(1)               BSD General Commands Manual              UUENCODE(1)

NAME
     uudecode, uuencode -- encode/decode a binary file

SYNOPSIS
     uuencode [-m] [-o output_file] [file] name
     uudecode [-cips] [file ...]
     uudecode [-i] -o output_file [file]

DESCRIPTION
     The uuencode and uudecode utilities are used to transmit binary files
     over transmission mediums that do not support other than simple ASCII
     data.

     The uuencode utility reads file (or by default the standard input) and
     writes an encoded version to the standard output, or output_file if one
     has been specified.  The encoding uses only printing ASCII characters and
     includes the mode of the file and the operand name for use by uudecode.

     The uudecode utility transforms uuencoded files (or by default, the stan-
     dard input) into the original form.  The resulting file is named either
     name or (depending on options passed to uudecode) output_file and will
     have the mode of the original file except that setuid and execute bits
     are not retained.  The uudecode utility ignores any leading and trailing
     lines.

     The following options are available for uuencode:

     -m     Use the Base64 method of encoding, rather than the traditional
            uuencode algorithm.

     -o output_file
            Output to output_file instead of standard output.

     The following options are available for uudecode:

     -c     Decode more than one uuencode'd file from file if possible.

     -i     Do not overwrite files.

     -o output_file
            Output to output_file instead of any pathname contained in the
            input data.

     -p     Decode file and write output to standard output.

     -s     Do not strip output pathname to base filename.  By default
            uudecode deletes any prefix ending with the last slash '/' for
            security purpose.

EXAMPLES
     The following example packages up a source tree, compresses it, uuencodes
     it and mails it to a user on another system.  When uudecode is run on the
     target system, the file ``src_tree.tar.Z'' will be created which may then
     be uncompressed and extracted into the original tree.

           tar cf - src_tree | compress |
           uuencode src_tree.tar.Z | mail sys1!sys2!user

     The following example unpack all uuencode'd files from your mailbox into
     your current working directory.

           uudecode -c < $MAIL

     The following example extract a compress'ed tar archive from your mailbox

           uudecode -o /dev/stdout < $MAIL | zcat | tar xfv -

LEGACY DESCRIPTION
     In legacy operation, uudecode masks file modes with 0666, preventing the
     creation of executable files.

     uudecode cannot change the mode of a created file which is not owned by
     the current user (unless that user is root).  In legacy operation,
     fchmod(2) allows the mode to be changed.

     For more information about legacy mode, see compat(5).

SEE ALSO
     basename(1), compress(1), mail(1), uucp(1), fchmod(2), uuencode(5)

BUGS
     Files encoded using the traditional algorithm are expanded by 35% (3
     bytes become 4, plus control information).

HISTORY
     The uudecode and uuencode utilities appeared in 4.0BSD.

BSD                            January 27, 2002                            BSD
```


</details>

Reply:
Wow, that's really old. On Linux, I can't even download `uuencode`. It's neither in the Manjaro repos, nor in the AUR. And if it's not in the AUR, it's questionable if it even exists 😆

Reply:
Try installing the package `sharutil` 🤔

Reply:
Found it. I searched for `shareutil` yesterday.